packages/cmr-client/src/EarthdataToken.ts

```diff
@@ -0,0 +1,183 @@
+  body: {
+    access_token?: string,
+    expiration_date?: string
+  }
+}>;
+
+export interface EarthdataTokenParams {
+  username: string,
+  password: string,
+  edlEnv: string,
+  token?: string,
+}
+
+/**
+ * A class to simplify requests for the Earthdata Login token used for CMR
+ *
+ * @typicalname earthdataToken
+ *
+ * @example
+ * const { EarthdataToken } = require('@cumulus/cmr-client');
+ *
+ * const earthdataToken = new EarthdataToken({
+ *  "username": "my-username",
+ *  "password": "my-password",
+ *  "edlEnv": "my-cmr-environment",
+ *  "token" : "my-token"
+ * });
+ */
+export class EarthdataToken {
+  username: string;
+  password: string;
+  edlEnv: string;
+  token?: string;
+
+  /**
+  * The constructor for the EarthdataToken class
+  *
+  * @param {string} params.username - Earthdata username
+  * @param {string} params.password - Earthdata password
+  * @param {string} params.edlEnv - the CMR environment of the user
+  * @param {string} params.token- the EarthdataLogin token, undefined if user doesn't have one
+  *
+  * @example
+  *
+  * {
+  *  "username": "janedoe",
+  *  "password": "password",
+  *  "edlEnv": "UAT",
+  *  "token" : "1782hg134bsd71"
+  *
+  * }
+  */
+
+  constructor(params: EarthdataTokenParams) {
+    this.username = params.username;
+    this.password = params.password;
+    this.edlEnv = params.edlEnv;
+    this.token = params.token;
+  }
+
+  getEDLurl(
+  ) {
+    switch (this.edlEnv) {
+      case 'PROD':
+      case 'OPS':
+        return 'https://urs.earthdata.nasa.gov';
+      case 'UAT':
+        return 'https://uat.urs.earthdata.nasa.gov';
+      case 'SIT':
+        return 'https://sit.urs.earthdata.nasa.gov';
+      default:
+        return 'https://sit.urs.earthdata.nasa.gov';
+    }
+  }
+
+  async getEDLToken(): Promise<string> {
+    if (!this.token) {
+      const buff = Buffer.from(`${this.username + ':' + this.password}`).toString('base64');
+      const url = this.getEDLurl();
+      // response: get a token from the Earthdata login endpoint using credentials if exists
+      let response: EarthdataGetTokenResponse;
+      try {
+        response = await got.get(`${url}/api/users/tokens`,
+          {
+            headers: {
+              Authorization: `Basic ${buff}`,
+            },
+          }).json();
+      } catch (error) {
+        const statusCode = get(error, 'response.statusCode', error.code);
+        const statusMessage = get(error, 'response.statusMessage', error.message);
+        let errorMessage = `Authentication error: Invalid Credentials, Authentication with Earthdata Login failed, statusCode: ${statusCode}, statusMessage: ${statusMessage}`;
+        const responseError = get(error, 'response.body.errors');
+        if (responseError) {
+          errorMessage = `${errorMessage}, CMR error message: ${JSON.stringify(responseError)}`;
+        }
+
+        throw new Error(errorMessage);
+      }
+      if (Object.keys(response).length === 0) {
```

Comment:
We should loop through the response and get the first unexpired token and return.  We can't assume how many tokens (2?) are there.

Reply:
We can't assume there are 1 or 2 EDL tokens, it might be 20 tokens.  We can just loop through the list and return the first valid one or undefined.